src/tests/dsETH.test.ts

```diff
@@ -1,55 +1,604 @@
+    const indexTokenAmount = wei('0.1')
+
+    await mint(outputToken, indexTokenAmount)
+  })
+
+  test('redeeming to ETH', async () => {
+    const inputToken = dsETH
+    const outputToken = ETH
+    const indexTokenAmount = wei('0.1')
+
+    await redeem(inputToken, indexTokenAmount)
+  })
+
+  test('minting with WETH', async () => {
+    const inputToken = WETH9
+    const outputToken = dsETH
+    const indexTokenAmount = wei('1')
+
+    const signer = SignerAccount0
+    await wrapETH(wei(2), signer)
+
+    await mintERC20(inputToken, outputToken, indexTokenAmount, 0.5, signer)
+  })
+  test('redeeming to WETH', async () => {
+    const inputToken = dsETH
+    const outputToken = WETH9
+    const indexTokenAmount = wei('1')
+
+    await redeemERC20(inputToken, outputToken, indexTokenAmount)
+  })
+
+  // FIXME: deposit into pool won't work because it reached max capacity
+  // test('minting with rETH', async () => {
+  //   const inputToken = RETH
+  //   const outputToken = dsETH
+  //   const indexTokenAmount = wei('0.1')
+
+  //   const signer = SignerAccount0
+  //   await depositIntoRocketPool(wei(2), signer)
+  //   await mintERC20(inputToken, outputToken, indexTokenAmount, 0.5, signer)
+  // })
+
+  test('redeeming to rETH', async () => {
+    // FIXME: remove later - for testing only, minting some dsETH
+    await mint(dsETH, wei('0.1'))
+
+    const inputToken = dsETH
+    const outputToken = RETH
+    const indexTokenAmount = wei('0.1')
+
+    await redeemERC20(inputToken, outputToken, indexTokenAmount)
+  })
+
+  test('minting with sETH2', async () => {
+    const inputToken = SETH2
+    const outputToken = dsETH
+    const indexTokenAmount = wei('1')
+
+    // ETH / sETH2
+    const signer = SignerAccount0
+    const pool = '0x7379e81228514a1d2a6cf7559203998e20598346'
+    await wrapETH(wei(2), signer)
+    await swapExactInput(
+      pool,
+      {
+        tokenIn: WETH9.address,
+        tokenOut: inputToken.address!,
+        amountIn: wei('2'),
+        amountOutMin: wei('1.5'),
+      },
+      provider,
+      signer
+    )
+
+    await mintERC20(inputToken, outputToken, indexTokenAmount, 0.5, signer)
+  })
+
+  test('redeeming to sETH2', async () => {
+    const inputToken = dsETH
+    const outputToken = SETH2
+    const indexTokenAmount = wei('0.1')
+
+    await redeemERC20(inputToken, outputToken, indexTokenAmount)
+  })
+
+  test('minting with stETH', async () => {
+    const inputToken = STETH
+    const outputToken = dsETH
+    const indexTokenAmount = wei('0.1')
+
+    const signer = SignerAccount0
+    await addLiquidityToLido(wei('1'), signer)
+
+    await mintERC20(inputToken, outputToken, indexTokenAmount, 0.5, signer)
+  })
+
+  test('redeeming to stETH', async () => {
+    const inputToken = dsETH
+    const outputToken = STETH
+    const indexTokenAmount = wei('0.1')
+
+    await redeemERC20(inputToken, outputToken, indexTokenAmount)
+  })
+
+  // FIXME: did work a few times already
```

Comment:
This test worked a few times already but now it usually fails with `uniswap/underbought` error.

Reply:
redeemERC20 is subject to slippage and price impact. Have you tried adding a very small slippage?